controllers/alphacontrollers/alert_controller.go

```diff
@@ -992,8 +995,17 @@ func flattenNotification(notification *alerts.AlertNotification) coralogixv1alph
 
 	switch integration := notification.GetIntegrationType().(type) {
 	case *alerts.AlertNotification_IntegrationId:
-		flattenedNotification.IntegrationID = new(int32)
-		*flattenedNotification.IntegrationID = int32(integration.IntegrationId.GetValue())
+		id := strconv.Itoa(int(integration.IntegrationId.GetValue()))
+		webhookStr, err := coralogixv1alpha1.WebhooksClient.GetWebhook(context.Background(), id)
+		if err != nil {
+			return flattenedNotification
+		}
+		var m map[string]interface{}
+		if err = json.Unmarshal([]byte(webhookStr), &m); err != nil {
```

Comment:
Also not sure about this error handling

Reply:
I think flattenNotification should return errors, and the error will be handled in the reconciliation loop

Reply:
The purpose of flatten is to initialize or update the status of the alert. I don't want to fail the update of the status just because of a failure in placing the integration-name. How about leaving the integration-name nil?

Reply:
If we leave the integration name as nil, the operator will try to reconcile it again since the Spec and Status are different, right? The loop will continue forever, until we don't fail to fetch a valid `alias`.

I think we want to fail the status update here, it will continue this reconciliation loop, but at least the metric `reconciliation_errors_total` provided by kubebuilder will increase, and we can trigger alerts for this scenario

Reply:
The reconcile happens forever anyway, the question is just in witch period.
If the Spec and Status are different it will try to update the alert with Update call to the backend but the period to the next reconciliation stays the same.

Reply:
Thanks for the explanation :) 

Still, I believe we want to fail here so we can monitor this failures with kubebuilder built-in metrics

Reply:
It makes the code little bit ugly :'(  but let me know if you think this is a good solution